src/renderer/store/Login.ts

```diff
@@ -0,0 +1,66 @@
+import { ipcRenderer } from 'electron'
+import axios from 'axios'
+import { Module, MutationTree, ActionTree } from 'vuex'
+
+export interface LoginState {
+  selectedInstance: string | null,
+  searching: boolean
+}
+
+const state = (): LoginState => ({
+  selectedInstance: null,
+  searching: false
+})
+
+export const MUTATION_TYPES = {
+  CHANGE_INSTANCE: 'changeInstance',
+  CHANGE_SEARCHING: 'changeSearching'
+}
+
+const mutations: MutationTree<LoginState> = {
+  [MUTATION_TYPES.CHANGE_INSTANCE]: (state: LoginState, instance: string) => {
+    state.selectedInstance = instance
+  },
+  [MUTATION_TYPES.CHANGE_SEARCHING]: (state: LoginState, searching: boolean) => {
+    state.searching = searching
+  }
+}
+
+// TODO: use type of rootState
+const actions: ActionTree<LoginState, any> = {
+  fetchLogin: (_, instance: string) => {
+    return new Promise((resolve, reject) => {
+      ipcRenderer.send('get-auth-url', instance)
+      ipcRenderer.once('error-get-auth-url', (_, err: Error) => {
+        ipcRenderer.removeAllListeners('response-get-auth-url')
+        reject(err)
+      })
+      ipcRenderer.once('response-get-auth-url', (_, url: string) => {
+        ipcRenderer.removeAllListeners('response-get-auth-url')
+        resolve(url)
+      })
+    })
+  },
+  pageBack: ({ commit }) => {
+    commit(MUTATION_TYPES.CHANGE_INSTANCE, null)
+  },
+  confirmInstance: async ({ commit }, domain: string): Promise<boolean> => {
+    commit(MUTATION_TYPES.CHANGE_SEARCHING, true)
+    await axios.get(`https://${domain}/api/v1/instance`)
+      .finally(() => {
+        commit(MUTATION_TYPES.CHANGE_SEARCHING, false)
+      })
+      commit(MUTATION_TYPES.CHANGE_INSTANCE, domain)
```

Comment:
__[ESLint]__ Error - Expected indentation of 4 spaces but found 6. (indent) ([link](https://sider.review/gh/repos/124234495/pulls/864#issue-3951122))

<sub>posted by Sider</sub>